internal/cmd/schema-registry/command_cluster_upgrade.go

```diff
@@ -35,36 +39,57 @@ func (c *command) newClusterUpgradeCommand() *cobra.Command {
 }
 
 func (c *command) clusterUpgrade(cmd *cobra.Command, _ []string) error {
-	packageDisplayName, err := cmd.Flags().GetString("package")
+	environmentId, err := c.Context.EnvironmentId()
 	if err != nil {
 		return err
 	}
 
-	packageInternalName, err := getPackageInternalName(packageDisplayName)
+	clusters, err := c.V2Client.GetSchemaRegistryClustersByEnvironment(environmentId)
 	if err != nil {
 		return err
 	}
 
-	environmentId, err := c.Context.EnvironmentId()
+	if len(clusters) == 0 {
+		return errors.NewSRNotEnabledError()
+	}
+
+	cluster := clusters[0]
+	clusterSpec := cluster.GetSpec()
+
+	packageDisplayName, err := cmd.Flags().GetString("package")
 	if err != nil {
 		return err
 	}
 
-	cluster, err := c.Context.FetchSchemaRegistryByEnvironmentId(environmentId)
+	_, err = getPackageInternalName(packageDisplayName)
 	if err != nil {
 		return err
 	}
 
-	if packageInternalName == cluster.Package {
+	if strings.EqualFold(clusterSpec.GetPackage(), packageDisplayName) {
 		output.ErrPrintf(errors.SRInvalidPackageUpgrade, environmentId, packageDisplayName)
 		return nil
 	}
-	cluster.Package = packageInternalName
 
-	if _, err := c.Client.SchemaRegistry.UpdateSchemaRegistryCluster(cluster); err != nil {
+	clusterUpdateRequest := createClusterUpdateRequest(packageDisplayName, environmentId)
+	_, err = c.V2Client.UpgradeSchemaRegistryCluster(*clusterUpdateRequest, cluster.GetId())
+	if err != nil {
 		return err
 	}
 
 	output.Printf(errors.SchemaRegistryClusterUpgradedMsg, environmentId, packageDisplayName)
 	return nil
 }
+
+func createClusterUpdateRequest(packageType, environmentId string) *srcmv2.SrcmV2ClusterUpdate {
```

Comment:
Can we create this struct manually instead of calling these functions? It's not clear what the "defaults" are, and we don't use this approach anywhere else.